src/conv/problem_description.cpp

```diff
@@ -59,6 +60,41 @@ std::ostream& operator<<(std::ostream& stream, std::function<void(std::ostream&)
     return stream;
 }
 
+void ProblemDescription::HeuristicUpdateLayouts()
+{
+    std::string labels = tensor_layout_get_default(in_layout.size());
+
+    std::set<std::string> input_layouts;
+    std::set<std::string> output_layouts;
+    std::set<std::string> weight_layouts;
+
+    std::vector<std::string> supported_layouts = {"NCHW", "NHWC", "NCDHW"};
+    for(const std::string& layout : supported_layouts)
+    {
+        // Skip layouts that doesn't match dimension sizes
+        if(layout.size() != labels.size())
+            continue;
+
+        if(in.IsLayout(labels, layout))
```

Comment:
`IsLayout` -> `IsLayoutPossible`  or `IsPossibleLayout`?

Reply:
[`IsLayout` ](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/765/files#diff-2a2c874ce9a2ebeea7603f4faa717453773c0ff152197455aa4ef0806df794ddR179) try to verify a exact match of the `layout` with `labels`, so I think the name here may be suitable

Reply:
Can a tensor have more than one layout?

Reply:
No, but here is just a simple check and I think the name `IsLayout` do meet its purpose.

Reply:
No? I thought that a tensor may match a number of layouts here. Am I wrong?

Reply:
Oh I think I got your idea. you mean here is just to check that this `TensorDescriptor` can be interpreted as the `labels` right? And this function's use scenario is for such a check, and maybe used multiple times.
Hi @jerryyin let me double confirm with you this function is designed for this purpose right?

Reply:
> Can a tensor have more than one layout?

@atamazov Correct, one tensor, when ambiguous, can have more than one layout. Here, it is checking for if a possible layout is coherent with its strides.

@carlushuang That's right, `IsPossibleLayout` is a better name, thanks.

Reply:
@jerryyin  OK, I'll change to this name